<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Bioinformatics learning</title>

<script src="site_libs/header-attrs-2.23/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">YH'Web</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="about.html">
    <span class="fa fa-info"></span>
     
    About
  </a>
</li>
<li>
  <a href="Bioinfo.html">
    <span class="fa fa-solid fa-dna"></span>
     
    Bioinformatics learning
  </a>
</li>
<li>
  <a href="Me.html">
    <span class="fa fa-regular fa-face-meh-blank"></span>
     
    Me
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-gear"></span>
     
    More
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Heading 1</li>
    <li>
      <a href="page-a.html">Page A</a>
    </li>
    <li>
      <a href="page-b.html">Page B</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Heading 2</li>
    <li>
      <a href="page-c.html">Page C</a>
    </li>
    <li>
      <a href="page-d.html">Page D</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://space.bilibili.com/314167138?spm_id_from=333.1007.0.0">
    <span class="fa fa-brands fa-bilibili"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Bioinformatics learning</h1>

</div>


<div id="r" class="section level1">
<h1>R</h1>
<div id="r基础知识" class="section level2">
<h2>R基础知识</h2>
<p><a
href="file:///Applications/RStudio.app/Contents/Resources/app/www/docs/keyboard.htm">RStudio快捷键</a></p>
<p><a href="https://rmarkdown.rstudio.com/lesson-1.html">R
markdown</a></p>
</div>
</div>
<div id="生信分析教程" class="section level1">
<h1>生信分析教程</h1>
<div id="atac-seq" class="section level2">
<h2>ATAC-seq</h2>
<div id="atac-seq-pipeline" class="section level3">
<h3><a href="https://www.hdsu.org/chipatac2020/">ATAC-seq
pipeline</a></h3>
<hr />
</div>
<div id="atac-seq-analysis" class="section level3">
<h3>ATAC-seq analysis</h3>
<div id="从华为云批量下载数据" class="section level4">
<h4>从华为云批量下载数据</h4>
<p><code>./obsutil share-cp authorization_code folder_url -r -f -j=20 -p=10 -vmd5 -ac=xxx</code></p>
<p>authorization_code即测序邮件提供的URL， fold_url即本地文件夹
-ac即提取码</p>
</div>
<div id="raw.data前处理" class="section level4">
<h4>Raw.data前处理</h4>
<div id="对rawread质检" class="section level5">
<h5>对RawRead质检</h5>
<p><code>fastqc -q -t 4 *gz -o ../02.raw._qc</code></p>
<p>-q安静运行模式，不报告实时运行情况，-t选择程序运行线程数，每个线程占用250MB内存</p>
</div>
<div id="去除接头-fastq-or-trim_galore" class="section level5">
<h5>去除接头: fastq or Trim_Galore</h5>
<p>fastq</p>
<pre><code> #! /bin/env perl
for fn in *R1.fastq.gz
do
     j = ${fn/R1.fastq.gz/}
    echo $fn, $j
    fastp -w 4 --adapter_fasta .../adaptors_all_fa绝对路径 --detect_adapter_for_pe -q 20 -l 18 -i $fn -I ${j}R2.fastq.gz -o ../03.clean_reads/${j}.qc1.fq.gz -O ../03.clean_reads/${j}.qc2.fq.gz
done</code></pre>
<p>Trim_Galore
<code>~/TrimGalore-0.6.6/trim_galore --phred33 --paired --fastqc --cores 3 --output_dir 输出路径 R1.fastq.gz R2.fastq.gz</code></p>
</div>
<div id="对clean_read质检" class="section level5">
<h5>对Clean_read质检</h5>
<p><code>fastqc -q -t 4 *fq.gz -o ../04.clean_qc</code></p>
</div>
<div id="将clean_read比对代参考基因组bowtie2" class="section level5">
<h5>将Clean_read比对代参考基因组：Bowtie2</h5>
<pre><code>#! bin/bash
for fn in *qc1.fq.gz
do
     j=${fn/qc1.fq.gz/}
    echo $fn, $j
    bowtie2 --very-sensitive -X 2000 -x .../tair10(参考基因组绝对路径) -1 ${j}qc1.fa.gz -2 ${j}qc2.fq.gz -S ../05.map/${j}.sam 2&gt; ../06.log/${j}map.log.txt
done| </code></pre>
</div>
<div id="去除线粒体叶绿体free-dnasamtoools" class="section level5">
<h5>去除线粒体、叶绿体free DNA：Samtoools</h5>
<pre><code>for i in *sam
do
     samtools view -@ 6 -bS $i &gt; ${i/sam}bam
done
for i in *bam
do
     samtools sort $i &gt; ${i/bam/}sort.bam
done
for i in *sort.bam
do
     samtools index $i
done
for i in *sort.bam
do
     samtools view -b $i Chr1 Chr2 Chr3 Chr4 Chr5 &gt; ${i/sort.bam/}rmMtPt.bam
done</code></pre>
<p>可选：统计mapped到染色体上的read数
<code>samtools idxstat -@ 3 inputfile.bam</code></p>
<p>统计到线粒体上的reads数
<code>samtools idxstat -@ 3 inputfile.bam | grep "chrM" | cut -f 3</code></p>
<p>统计mapped到所有染色体的read数
<code>samtools idxstat -@ 3 inputfile.bam | awk '{SUM += $3} END {prints SUM}'</code></p>
</div>
<div id="去除duplicate-readspicard-markduplicates"
class="section level5">
<h5>去除Duplicate reads：Picard MarkDuplicates</h5>
<pre><code>for i in *rmMtPt.bam
do
     java -jar .../picard.jar绝对路径 MarkDupicates I=$i O=${i/rmMtPt/}rmd.bam ASSUE_SORTED=ture REMOVE_DUPLICATES=ture M=${i/rmMtPt.bam/}rmd_metrics.txt
done
for i in *rmd.bam
do
     samtools view -bq 30 $i &gt; ${i/rmd.bam}rmMulti.sort.bam
done
for i in *rmMulti.sort.bam
do
     samtools index -@ 6 $i
done</code></pre>
</div>
<div id="可选nucleosome-positioning-r-package-libraryatacseqqc"
class="section level5">
<h5>可选：Nucleosome positioning: R package: library(“ATACseqQC”)</h5>
<p><code>fragsize &lt;- fragSizeDist(bamFiles = "rmDup.bam绝对路径", index = "rmDup.bam绝对路径, bamFiles.labels = "nucleosome positioning")</code>
<img src="photo/Nocleosome%20positioning.jpg" alt="caption" /></p>
</div>
<div id="atac-shift-deeptools-alignmentsieve" class="section level5">
<h5>ATAC shift: Deeptools alignmentSieve</h5>
<pre><code>for i in *rmMulti.sort.bam
do
      alignmentSieve --numberOfProcessors 8 --ATACshift --bam $i -o ${i/rmMulti.sort.bam}shift.sort.bam
done</code></pre>
</div>
<div id="peak-caller-macs2-callpeak" class="section level5">
<h5>Peak caller: MACS2 callpeak</h5>
<pre><code>for i in *shift.sort.bam
do
     macs2 callpeak -t $i -f BAMPE -g 1.2e8 --keep-dup all --cutoff-analysis -n ${i/.shift.sort.bam} --outdir ../07.peak/${i/.sort.bam} 2&gt; ../06.log/${i/sort.bam}macs2.log.txt
done &gt; ../06.log/jim.shift.macs2.log.txt 2&gt;&amp;1
    </code></pre>
<p>-t –treatment FILENAME: input file direction</p>
<p>-c –control: the control, genomic input or mock IP data file</p>
<p>-n –name: the name of experiment</p>
<p>–outdir: MACS will save all output files into the specified folder
for this option</p>
<p>-f –format FORMAT: BAMPE and BEDPE need to be specified</p>
<p>-g –gsize: it’s the mappable genome size or effective genome size
which is defined as the genome size which can be sequenced.
拟南芥实际基因组大小1.2e8（一亿两千万）</p>
<p>–keep-dup: It controls the MACS behavior towards duplicate tags at
the exact same location</p>
</div>
<div id="deeptools-bamcoverage" class="section level5">
<h5>Deeptools: bamCoverage</h5>
<pre><code>for i in*shift.sort.bam
do
     bamCoverage --numberprocessors 6 --binsize 10 --nomalizeUsing RPGC --effectiveGenomeSieve 120000000 --bam $i -o ../07.peak/${i/sort.bam}bw
done</code></pre>
</div>
<div id="plot-correlation-deeptools-multibamsummary-and-plotcorrelation"
class="section level5">
<h5>Plot correlation: Deeptools: multiBamSummary and
plotCorrelation</h5>
<pre><code>multiBamSummary bins --bamfiles *bam -o ../08.plot/cov.results.npz
plotCorrelation -in ../08.plot/cov.results.npz --corMethod spearman --skipZeros --plotTitle &quot;Spearman Correlation of Read Counts&quot; --whatToplot heatmap --colorMap RdYlbu --plotNumbers -o ../08.plot/heatmap_SpearmanCorr_readCounts.png --outFileCorMatrix ../08.plot/SpearmanCorr_readcounts.tab</code></pre>
</div>
<div
id="plot-tss-enrich-heatmap-deeptools-computematrix-and-plotheatmap"
class="section level5">
<h5>Plot TSS enrich heatmap: Deeptools: computeMatrix and
plotHeatmap</h5>
<pre><code>for i in *bw
do
     computematrix scale-regions -p 6 -S $i -R ..../Araport11_whole_gene_for_deeptools.bed --regionBodyLength 2000 --beforeRegionStartLength 3000 --afterRegionStartLength 3000 --skipZeros -o${i/bwt2.rmMutli.bw/}matrix_scale_region.gz
done
for i in*gz
do
     plotHeatmap -m $i -out ${i/.matrix_scale_region.scale.gz/}.png --colorMap YlGnBu --regionsLabel ${i/./matrix_scale_region.scale.gz/} --heatmapHeight 15 --plotTitle ${i/.matrix_scale_regions.scale.gz/}
done</code></pre>
<div class="float">
<img src="photo/TSS%20enrich%20plot.png" alt="caption" />
<div class="figcaption">caption</div>
</div>
</div>
</div>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
